pinot-plugins/pinot-input-format/pinot-csv/src/test/java/org/apache/pinot/plugin/inputformat/csv/CSVRecordReaderTest.java

```diff
@@ -216,4 +219,555 @@ public void testNullValueString()
     Assert.assertNull(row.getValue("col2"));
     Assert.assertNull(row.getValue("col3"));
   }
+
+  @Test
```

Comment:
Instead of repeating entire function for default/line iterator, we can keep 1 test but within each test we can test both. For all existing tests (e.g. `testNullValueString`, `testHeaderDelimiterSingleColumn` etc), let's add the coverage for line iterator as well.

I think that we can make this test very concise like the following:

```
    URI uri = ClassLoader.getSystemResource("dataFileWithCommentedLines.csv").toURI();
    File dataFile = new File(uri);

    CSVRecordReaderConfig readerConfig = new CSVRecordReaderConfig();
    readerConfig.setSkipUnParseableLines(true);
    readerConfig.setCommentMarker('#');
    List<GenericRow> genericRows = readCSVRecords(dataFile, readerConfig);
    Assert.assertEquals(3, genericRows.size());
    
    readerConfig.setSkipUnParseableLines(true);
    genericRows = readCSVRecords(dataFile, readerConfig);
    Assert.assertEquals(3, genericRows.size());
```